ExtCore/Pervasive.fs

```diff
@@ -1012,188 +1034,141 @@ module Option =
-            Choice1Of2 result
+            Ok result
         | None ->
-            Choice2Of2 ()
+            Error ()
+
+    /// <summary></summary>
+    /// <param name="value"></param>
+    /// <returns></returns>
+    [<CompiledName("OfChoice")>]
+    let ofChoice (value : Choice<'T,'Error>) : Result<'T, 'Error> =
+        match value with
+        | Choice1Of2 result ->
+            Ok result
+        | Choice2Of2 err ->
+            Error err
 
     /// <summary></summary>
     /// <param name="errorValue"></param>
     /// <param name="value"></param>
     /// <returns></returns>
-    // TODO :   Rename this to 'ofOptionDefault' or 'ofOptionWithDefault'.
-    //          The "With" suffix should be reserved for higher-order functions.
     [<CompiledName("OfOptionWith")>]
-    let ofOptionWith (errorValue : 'Error) (value : 'T option) : Choice<'T, 'Error> =
+    let ofOptionDefault (errorValue : 'Error) (value : 'T option) : Result<'T, 'Error> =
         match value with
         | Some result ->
-            Choice1Of2 result
+            Ok result
         | None ->
-            Choice2Of2 errorValue
+            Error errorValue
 
     /// <summary></summary>
     /// <param name="value"></param>
     /// <returns></returns>
     [<CompiledName("ToOption")>]
-    let toOption (value : Choice<'T, 'Error>) : 'T option =
-        // Preconditions
-        checkNonNull "value" value
-
+    let toOption (value : Result<'T, 'Error>) : 'T option =
         match value with
-        | Choice1Of2 result ->
+        | Ok result ->
             Some result
-        | Choice2Of2 _ ->
+        | Error _ ->
             None
 
-    /// <summary>
-    /// When the choice value is <c>Choice1Of2(x)</c>, returns <c>Choice1Of2 (f x)</c>.
-    /// Otherwise, when the choice value is <c>Choice2Of2(x)</c>, returns <c>Choice2Of2(x)</c>.
-    /// </summary>
-    /// <param name="mapping"></param>
-    /// <param name="value"></param>
-    /// <returns></returns>
-    [<CompiledName("Map")>]
-    let map (mapping : 'T -> 'U) (value : Choice<'T, 'Error>) =
-        // Preconditions
-        checkNonNull "value" value
-
-        match value with
-        | Choice1Of2 result ->
-            Choice1Of2 (mapping result)
-        | Choice2Of2 error ->
-            Choice2Of2 error
-
-    /// <summary>
-    /// Applies the specified mapping function to a choice value representing an error value (Choice2Of2). If the choice
-    /// value represents a result value (Choice1Of2), the result value is passed through without modification.
-    /// </summary>
-    /// <param name="mapping"></param>
+    /// <summary></summary>
     /// <param name="value"></param>
     /// <returns></returns>
-    [<CompiledName("MapError")>]
-    let mapError (mapping : 'Error1 -> 'Error2) (value : Choice<'T, 'Error1>) =
-        // Preconditions
-        checkNonNull "value" value
-
+    [<CompiledName("ToChoice")>]
+    let toChoice (value : Result<'T, 'Error>) : Choice<'T,'Error> =
```

Comment:
Same, do we need it?

Reply:
In order to be compatible with libraries that does not implement `Result` ?